module-3/utils.js

```diff
@@ -0,0 +1,160 @@
+const fs = require('fs');
+const path = require('path');
+const stream = require('stream');
+const csvParser = require('csv-parse');
+const lockfile = require('proper-lockfile');
+
+const filePath = path.join('.', 'events.csv');
+
+const lockCsv = () => lockfile.lock(filePath, {
+  retries: {
+    retries: 500,
+    factor: 3,
+    minTimeout: 10,
+    maxTimeout: 60 * 1000,
+    randomize: true
+  }
+});
+
+const csvParserConfig = {
+  delimiter: ',',
+  from_line: 1,
+  columns: true,
+  skip_empty_lines: true
+}
+
+const headers = "id,title,location,date,hour";
+const getCsvItem = ({id = '', title = '', location = '', date = '', hour = ''}) => `${id},${title},${location},${date},${hour}`
+const transformEventsToCsv = (events) => {
+  const lineSeparator = "\n";
+  const data = events.map(getCsvItem);
+  return `${headers}${lineSeparator}${data.join(lineSeparator)}${lineSeparator}`;
+}
+
+const addNewEvent = async event => {
+  const unlock = await lockCsv();
+  return new Promise((resolve, reject) => {
+    const stream = fs.createWriteStream(filePath, {flags: 'a'});
+    stream.on('error', (err) => {
+      reject(err);
+      unlock();
+    })
+    stream.on('finish', () => {
+      resolve(event);
+      unlock();
+    })
+    stream.write(getCsvItem(event) + '\n');
+    stream.end();
+  });
+}
+
+const getEvents = async () => {
+  return new Promise((resolve) => {
+    fs.readFile(filePath, (err, data) => {
+      if (err) throw err;
```

Comment:
Instead of throw we may have to use `reject(err)` here since we are creating our own promise here.